test/jit/test_class_type.py

```diff
@@ -1180,6 +1180,15 @@ def attr(self) -> int:
             def unsupported(self) -> int:
                 return sum([self.a])
 
+            @property
```

Comment:
Do you know would it take to support the unused decorator. and then property ? That seems more intuitive to me. With the current API I would expect the property to exist but the getter be an ignored function, whereas we're just completely dropping it

```
            @torch.jit.unused
            @property
            def unsupported_2(self) -> int:
                return sum([self.a])

            @unsupported_2.setter
            def unsupported_2(self, value):
                self.a = sum([self.a])
```

Reply:
So this is the thing that I tried to do in my diff for properties on modules. But there are backwards compatibility issues because using `@unused` or `@ignored` like this produces an error in older versions of PyTorch.

The concrete case we ran into is that some internal library is using pytorch 1.4 in their OSS tree and pytorch from master in fbcode.

Reply:
Trying to make new features & syntax work across older versions of pytorch is going to limit the clarity and intuitiveness of the APIs we make. it's also not a very common use case. It's good to offer escape hatches here (like __jit_ignored_attributes__) but I don't think we should overly taylor APIs to it

Reply:
Oh, I thought you were meant that we should introduce a decorator-based syntax and get rid of the `__ignored_attributes__` escape hatch. But if we're going to have both, I agree that it's better to allow people to write:
```
@torch.jit.unused
@property 
...
```

I'll update the PR.